src/utils/style-formatter.js

```diff
@@ -2,71 +2,39 @@ import colorUtils from './color-utils';
 import luiIcons from './lui-icons';
 import urlUtils from './url-utils';
 import DEFAULTS from '../style-defaults';
-
-const backgroundSize = {
-  auto: 'auto auto',
-  alwaysFit: 'contain',
-  fitWidth: '100% auto',
-  fitHeight: 'auto 100%',
-  fill: '100% 100%',
-  alwaysFill: 'cover',
-};
-
-const backgroundPosition = {
-  topLeft: '0% 0%', // top left
-  centerLeft: '50% 0%', // center left
-  bottomLeft: '100% 0%', // bottom left
-  topCenter: '0% 50%', // top center
-  centerCenter: '50% 50%', // center center
-  bottomCenter: '100% 50%', // bottom center
-  topRight: '0% 100%', // top right
-  centerRight: '50% 100%', // center right
-  bottomRight: '100% 100%', // bottom right
-};
+import { getColor, backgroundSize, backgroundPosition } from './style-utils';
 
 const formatProperty = (path, setting) => `${path}: ${setting};`;
 
-const getColor = (
-  { useColorExpression = false, colorExpression = '', color = DEFAULTS.COLOR },
-  defaultColor,
-  theme
-) => {
-  let resolvedColor;
-  if (useColorExpression) {
-    resolvedColor = colorUtils.resolveExpression(colorExpression);
-  } else if (typeof color === 'string') {
-    resolvedColor = color;
-  } else if (color) {
-    resolvedColor = theme.getColorPickerColor(color);
-  }
-  return !resolvedColor || resolvedColor === 'none' ? defaultColor : resolvedColor;
-};
-
 export default {
   getStyles({ style = {}, disabled, theme, element, app }) {
     let styles =
       'width: 100%;height: 100%;transition: transform .1s ease-in-out;position: absolute;bottom: 0;left: 0; top: 0;right: 0;margin: auto;';
     const { font = {}, background = {}, border = {} } = style;
-    const primaryColor = theme.getDataColorSpecials().primary;
     // enable
     styles += disabled ? formatProperty('opacity', 0.4) : formatProperty('cursor', 'pointer');
-    const backgroundColor = getColor(background, primaryColor, theme);
     // font
-    styles += formatProperty('color', getColor(font, '#ffffff', theme));
+    styles += formatProperty('color', getColor(font, DEFAULTS.FONT_COLOR.color, theme));
     const fontStyle = font.style || DEFAULTS.FONT_STYLE;
     fontStyle.bold && (styles += formatProperty('font-weight', 'bold'));
     fontStyle.italic && (styles += formatProperty('font-style', 'italic'));
     // background
+    const primaryColor = theme.getDataColorSpecials().primary;
+    const backgroundColor = getColor(background, primaryColor, theme);
     styles += formatProperty('background-color', backgroundColor);
-    if (background.useImage && background.url.qStaticContentUrl) {
+    // backgroundImage
+    const {useImage} = background;
```

Comment:
Are you using the linter? It should do this automatically
```suggestion
    const { useImage } = background;
```